crates/biome_js_analyze/src/lint/nursery/use_focusable_interactive.rs

```diff
@@ -0,0 +1,108 @@
+use crate::services::aria::Aria;
+
+declare_rule! {
+    /// Elements with an interactive role and interaction handlers must be focusable.
+    ///
+    /// HTML elements with interactive roles must have `tabIndex` defined to ensure they are
+    /// focusable. Without tabIndex, assistive technologies may not recognize these elements as
+    /// interactive.
+    /// You could also consider switching from an interactive role to its semantic HTML element
+    /// instead.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <div role="button" />
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// <div role="tab" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <div role="button" tabIndex={0} />
+    /// ```
+    ///
+    /// ```jsx
+    /// <div />
+    /// ```
+    ///
+    pub UseFocusableInteractive {
+        version: "next",
+        name: "useFocusableInteractive",
+        sources: &[RuleSource::EslintJsxA11y("interactive-support-focus")],
+        recommended: true,
+    }
+}
+
+pub struct RuleState {
+    role_name: String,
+}
+
+impl Rule for UseFocusableInteractive {
+    type Query = Aria<AnyJsxElement>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        if !node.is_element() {
+            return None;
+        }
+
+        let element_name = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+        let aria_roles = ctx.aria_roles();
+        let attributes = ctx.extract_attributes(&node.attributes());
+
+        if aria_roles.is_not_interactive_element(element_name.text_trimmed(), attributes) {
+            let role_attribute = node.find_attribute_by_name("role");
+            if let Some(role_attribute) = role_attribute {
+                let tabindex_attribute = node.find_attribute_by_name("tabIndex");
+                let role_attribute_value = role_attribute.initializer()?.value().ok()?;
+                if attribute_has_interactive_role(&role_attribute_value, aria_roles)?
+                    && tabindex_attribute.is_none()
+                {
+                    return Some(RuleState {
+                        role_name: role_attribute_value.text(),
+                    });
+                }
+            }
+        }
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "The HTML element with the interactive role "<Emphasis>{state.role_name}</Emphasis>" is not focusable."
+                },
+            )
+            .note(markup! {
+                "Adding a tabIndex attribute will make this element focusable."
+            }),
```

Comment:
Some notes: 
- I think the error message should be more explicit, and it should focus on the fact that interactive roles must be focusable 
- Based on our rule pillars, we need to explain the error to the user. I.e. why the absence of tab index is an error 
- Is the `.note` a suggestion to fix the issue? If so, I would phrase it with a direct tone e.g. "Add the tabIndex attribute.". We could put an emphasis on `tabIndex`